local/src/core/ProxyVerifier.cc

```diff
@@ -2030,6 +2116,260 @@ AbsenceCheck::test(swoc::TextView key, swoc::TextView name, std::list<swoc::Text
   return true;
 }
 
+bool
+ContainsCheck::test(swoc::TextView key, swoc::TextView name, swoc::TextView value) const
+{
+  swoc::Errata errata;
+  if (name.empty())
+    errata.info(
+        R"(Contains Violation: Absent. Key: "{}", Name: "{}", Required Value: "{}")",
+        key,
+        _name,
+        _value);
+  else if (value.find(_value) == std::string::npos)
+    errata.info(
+        R"(Contains Violation: Not Contained. Key: "{}", Name: "{}", Required Value: "{}", Actual Value: "{}")",
+        key,
+        _name,
+        _value,
+        value);
+  else {
+    errata.info(
+        R"(Contains Success: Key: "{}", Name: "{}", Required Value: "{}", Value: "{}")",
+        key,
+        _name,
+        _value,
+        value);
+    return true;
+  }
+  return false;
+}
+
+bool
+ContainsCheck::test(
+    swoc::TextView key,
+    swoc::TextView name,
+    std::list<swoc::TextView> const &values) const
+{
+  swoc::Errata errata;
+  if (name.empty() || values.size() != _values.size()) {
+    MSG_BUFF message;
+    message.print(R"(Contains Violation: Absent/Mismatched. Key: "{}", Name: "{}", )", key, _name);
+    message.print(R"(Required Values:)");
+    for (auto const &value : _values) {
+      message.print(R"( "{}")", value);
+    }
+    message.print(R"(, Received Values:)");
+    for (auto const &value : values) {
+      message.print(R"( "{}")", value);
+    }
+    errata.info(message.view());
+    return false;
+  }
+  auto it = values.begin();
+  auto jt = _values.begin();
```

Comment:
I suggest: expecte_it, found_it

(There's a few other places this would apply to.)

Reply:
Heh. I use a trailing "_p" for "predicate" for these types of things. E.g.
```
bool expect_p = false, found_p = true;
```
But that's personal style.